cli/src/main/java/software/amazon/smithy/java/cli/SmithyCall.java

```diff
@@ -0,0 +1,297 @@
+import picocli.CommandLine.ArgGroup;
+import software.amazon.smithy.java.aws.client.auth.scheme.sigv4.SigV4AuthScheme;
+import software.amazon.smithy.java.aws.client.awsjson.AwsJson1Protocol;
+import software.amazon.smithy.java.aws.client.core.identity.EnvironmentVariableIdentityResolver;
+import software.amazon.smithy.java.aws.client.core.settings.RegionSetting;
+import software.amazon.smithy.java.aws.client.restjson.RestJsonClientProtocol;
+import software.amazon.smithy.java.aws.client.restxml.RestXmlClientProtocol;
+import software.amazon.smithy.java.client.core.auth.scheme.AuthSchemeResolver;
+import software.amazon.smithy.java.client.core.endpoint.EndpointResolver;
+import software.amazon.smithy.java.client.core.interceptors.ClientInterceptor;
+import software.amazon.smithy.java.client.core.interceptors.RequestHook;
+import software.amazon.smithy.java.client.rpcv2.RpcV2CborProtocol;
+import software.amazon.smithy.java.core.serde.ShapeSerializer;
+import software.amazon.smithy.java.core.serde.document.Document;
+import software.amazon.smithy.java.dynamicclient.DynamicClient;
+import software.amazon.smithy.java.json.JsonCodec;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.shapes.OperationShape;
+import software.amazon.smithy.model.shapes.ShapeId;
+
+@Command(name = "smithy-call", mixinStandardHelpOptions = true, version = "1.0",
+        description = "Smithy Java CLI")
+public final class SmithyCall implements Callable<Integer> {
+    private static final JsonCodec CODEC = JsonCodec.builder().build();
+    private static final Logger LOGGER = Logger.getLogger(SmithyCall.class.getName());
+
+    private static final String[] BASE_RESOURCE_FILES = {
+            "aws.api.smithy",
+            "aws.auth.smithy",
+            "aws.customizations.smithy",
+            "aws.protocols.smithy"
+    };
+
+    @Parameters(index = "0", description = "Service Name")
+    private String service;
+
+    @Parameters(index = "1", description = "Name of the operation to perform on the service", arity = "0..1")
+    private String operation;
+
+    @Option(names = {"-v", "--verbose"}, description = "Enable verbose logging")
+    private boolean verbose;
+
+    @Option(names = { "-m", "--model-path" }, description = "Path to a directory containing all necessary .smithy service model files", required = true)
+    private String[] modelPath;
+
+    @Option(names = "--input-path", description = "Path to a JSON file containing input parameters for the operation")
+    private String inputPath;
+
+    @Option(names = "--input-json", description = "JSON string containing input parameters for the operation")
+    private String inputJson;
+
+    @Option(names = "--url", description = "Endpoint URL for the service")
+    private String url;
+
+    @Option(names = { "-p", "--protocol" }, description = "Communication protocol to use (options: awsjson, rpcv2-cbor, restjson, and restxml)")
+    private String protocol;
```

Comment:
You can directly use your `ProtocolType` enum here and [picocli will automatically display its possible values and enable command-line completion for them](https://picocli.info/#_enum_types)